### PR TITLE
fix: set key in ruler configmap

### PR DESCRIPTION
**Pull request by abhishiekgowtham**
when trying to deploy the thanos ruler with rules enabled without this change, got this error from k8s-apiserver

`Error: error validating "": error validating data: ValidationError(ConfigMap.data.groups): invalid type for io.k8s.api.core.v1.ConfigMap.data: got "array", expected "string"`

This PR sets the key for the configmap and key name can be anything or made as a helm chart variable since the ruler can read any `yaml` files in that directory https://github.com/stevehipwell/helm-charts/blob/main/charts/thanos/templates/rule/statefulset.yaml#L90

### CODE REVIEW

**Comment by stevehipwell** [comment]
Thanks for the PR @abhishiekgowtham. Could you update the CHANGELOG to cover your change (you can keep it as unreleased see below)?

```markdown
## [UNRELEASED]

### All Changes

- Fixed issue in Rule config map.
```

**Comment by abhishiekgowtham** [comment]
`CHANGELOG.md` is updated with the change @stevehipwell

**Review by onedr0p — APPROVED** [review]

File: charts/thanos/CHANGELOG.md [hunk 1]
```diff
@@ -14,9 +14,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Updated
 - Changed
 - Fixed
+  - issue in Rule config map.
```
Comment (stevehipwell) [inline charts/thanos/CHANGELOG.md:17]: Could you remove this line?

**Review by stevehipwell — CHANGES_REQUESTED** [review]
If you can remove the erroneous line in the CHANGELOG I'll get this merged and released with the latest Thanos version.

File: charts/thanos/CHANGELOG.md (hunk 1 above)
Comment (abhishiekgowtham) [reply charts/thanos/CHANGELOG.md]: fixed in https://github.com/stevehipwell/helm-charts/pull/641/commits/08424dcc0f18956247d926f16a94b5f55099f943

**Review by stevehipwell — APPROVED** [review]
LGTM

**Pull request merged**